Userland/Applications/Browser/Tab.cpp

```diff
@@ -57,12 +57,14 @@ URL url_from_user_input(const String& input)
 
 void Tab::start_download(const URL& url)
 {
-    auto window = GUI::Window::construct(&this->window());
-    window->resize(300, 170);
-    window->set_title(String::formatted("0% of {}", url.basename()));
-    window->set_resizable(false);
-    window->set_main_widget<DownloadWidget>(url);
-    window->show();
+    // URL basenames are not this consistent :-(
+    // TODO: More work identifying potential file names
+    auto filename_parts = url.basename().split('.');
+
+    if (filename_parts.size() == 2)
+        m_filesystem_client->async_prompt_download_file(url, filename_parts[0], filename_parts[1]);
+    else
+        m_filesystem_client->async_prompt_download_file(url, url.basename(), "txt");
```

Comment:
You can either find the last `.` in the URL and split on that one manually, or use `LexicalPath`:
```suggestion
auto filename = LexicalPath(url.basename());
auto extension = filename.extension();
if (extension.is_empty())
    extension = "txt";
m_filesystem_client->async_prompt_download_file(url, filename.title(), extension);
```